atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/AbstractSweepQueueTest.java

```diff
@@ -150,16 +155,24 @@ void putTimestampIntoTransactionTable(long ts, long commitTs) {
     List<WriteInfo> writeToCellsInFixedShard(SweepQueueTable writer, long ts, int number, TableReference tableRef) {
         List<WriteInfo> result = new ArrayList<>();
         for (long i = 0; i < number; i++) {
-            Cell cell = getCellWithFixedHash(i);
-            result.add(WriteInfo.write(tableRef, cell, ts));
+            CellReference cellRef = getCellRefWithFixedShard(i, tableRef, numShards);
+            result.add(WriteInfo.write(tableRef, cellRef.cell(), ts));
         }
         putTimestampIntoTransactionTable(ts, ts);
         writer.enqueue(result);
         return result;
     }
 
-    static Cell getCellWithFixedHash(long seed) {
-        return Cell.create(PtBytes.toBytes(seed), PtBytes.toBytes(seed));
+    static CellReference getCellRefWithFixedShard(long seed, TableReference tableRef, int shards) {
+        byte[] rowName = PtBytes.toBytes(seed);
+        int counter = 0;
+        CellReference cellReference;
+        do {
+            byte[] colName = PtBytes.toBytes(counter);
+            cellReference = CellReference.of(tableRef, Cell.create(rowName, colName));
+            counter++;
+        } while (IntMath.mod(cellReference.hashCode(), shards) != 0);
```

Comment:
Is it a problem that this function as written becomes more expensive as the number of shards increases?